micropython/build.sh

```diff
@@ -21,24 +24,55 @@ mkdir -p "$PREFIX_UPYTH_BUILD" "$PREFIX_UPYTH_MARKERS"
 
 
 #
-# Apply patches
+# Apply patches and copy files
 #
 # FIXME: prefix patch files with numbers to ensure correct order
-for patchfile in "$PREFIX_UPYTH_PATCHSET"/*.patch; do
+for patchfile in "$PREFIX_UPYTH_CONFIG"/patches/*.patch; do
 	if [ ! -f "$PREFIX_UPYTH_MARKERS/$(basename "$patchfile").applied" ]; then
 		echo "applying patch: $patchfile"
 		patch -d "$PREFIX_UPYTH_SRC" -p1 < "$patchfile"   # FIXME: should be -p1
 		touch "$PREFIX_UPYTH_MARKERS/$(basename "$patchfile").applied"
 	fi
 done
+cp -a "${PREFIX_UPYTH_CONFIG}/files/001_mpconfigport.mk" "${PREFIX_UPYTH_SRC}/ports/unix/mpconfigport.mk" && echo "Copied mpconfigport.mk!"
+
+
+#
+# Micropython internal use stack/heap size (not actual application stack/heap)
+# May be overwritten in _targets/build.project.*
+#
+if [[ ! $UPYTH_STACKSZ ]]; then
+	UPYTH_STACKSZ="4096"
+fi
+if [[ ! $UPYTH_HEAPSZ ]]; then
+	UPYTH_HEAPSZ="16384"
+fi
+
+
+#
+# Architecture specific flags/values set
+#
+if [ "${TARGET_FAMILY}" = "armv7m7" ]
+then
+	export LDFLAGS_EXTRA=" -Wl,-q ${CFLAGS}"
+	export STRIPFLAGS_EXTRA="--strip-unneeded"
+elif [ "${TARGET_FAMILY}" = "ia32" ]
+then
+	export STRIPFLAGS_EXTRA="--strip-all"
+else
+	echo "Error: there is no micropython port for chosen architecture"
+	exit 1
+fi
+export PHOENIX_MATH_ABSENT="expm1 log1p asinh acosh atanh erf tgamma lgamma copysign __sin __cos __tan __signbit"
+export CFLAGS_EXTRA="${CFLAGS} -DUPYTH_STACKSZ=${UPYTH_STACKSZ} -DUPYTH_HEAPSZ=${UPYTH_HEAPSZ} "
+
 
 #
 # Build and install micropython binary
 #
-# FIXME - remove clean
-(cd "${PREFIX_UPYTH_SRC}/mpy-cross" && make clean && make all BUILD="${PREFIX_UPYTH_BUILD}")
-(cd "${PREFIX_UPYTH_SRC}/ports/unix" && make clean && make all)
+(cd "${PREFIX_UPYTH_SRC}/mpy-cross" && make all BUILD="${PREFIX_UPYTH_BUILD}" CROSS_COMPILE="${CROSS}")
+(cd "${PREFIX_UPYTH_SRC}/ports/unix" && make all CROSS_COMPILE="${CROSS}")
 
 cp -a "${PREFIX_UPYTH_SRC}/ports/unix/micropython" "$PREFIX_PROG"
```

Comment:
nah, no unstripped version available, so don't put anything in `PREFIX_PROG` (unless You can get the unstripped version somehow)

Reply:
Copying to prog removed